homeassistant/components/konnected/config_flow.py

```diff
@@ -0,0 +1,732 @@
+                    description_placeholders={
+                        "zone": "Zone " + self.active_cfg
+                        if len(self.active_cfg) < 3
+                        else self.active_cfg.upper()
+                    },
+                    errors=errors,
+                )
+
+        return await self.async_step_options_switch()
+
+    async def async_step_options_switch(self, user_input=None):
+        """Allow the user to configure the IO options for switches."""
+        errors = {}
+        if user_input is not None:
+            zone = {"zone": self.active_cfg}
+            zone.update(user_input)
+            self.switches.append(zone)
+            self.io_cfg.pop(self.active_cfg)
+            self.active_cfg = None
+
+        if self.active_cfg:
+            return self.async_show_form(
+                step_id="options_switch",
+                data_schema=vol.Schema(DATA_SCHEMA_SWITCH_OPTIONS),
+                description_placeholders={
+                    "zone": "Zone " + self.active_cfg
+                    if len(self.active_cfg) < 3
+                    else self.active_cfg.upper()
+                },
+                errors=errors,
+            )
+
+        # find the next unconfigured binary sensor
+        for key, value in self.io_cfg.items():
+            if value == "Switchable Output":
+                self.active_cfg = key
+                return self.async_show_form(
+                    step_id="options_switch",
+                    data_schema=vol.Schema(DATA_SCHEMA_SWITCH_OPTIONS),
+                    description_placeholders={
+                        "zone": "Zone " + self.active_cfg
+                        if len(self.active_cfg) < 3
+                        else self.active_cfg.upper()
+                    },
+                    errors=errors,
+                )
+
+        # Build a config mimicking configuration.yaml
+        return await self.async_create_or_update_entry(
+            {
+                "host": self.host,
+                "port": self.port,
+                "id": self.device_id,
+                "binary_sensors": self.binary_sensors,
+                "sensors": self.sensors,
+                "switches": self.switches,
+            }
+        )
+
+    async def async_create_or_update_entry(self, device_config):
+        """Create or update a config entry based on the config flow info.
+
+        If an existing config entry is found, we will validate the info
+        and replace the entry. Otherwise we will create a new one.
+        """
+        try:
+            device_config = DEVICE_SCHEMA(device_config)
+
+        except vol.Invalid as err:
+            _LOGGER.error(
+                "Invalid device config..%s", humanize_error(device_config, err),
+            )
+            return self.async_abort(reason="unknown")
+
+        # Remove all other entries of panels with same ID or host
+        same_panel_entries = [
```

Comment:
This will probably not be needed after using the config entry unique_id api.

Reply:
👍